src/Bowling.js

```diff
@@ -0,0 +1,121 @@
+var Game = function() {
+  this.score = 0;
+  this.frame = 0;
+  this.frameBall = 1;
+  this.frameScore = 0;
+  this.scoreCard = [];
+  this.pinsInPlay = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+  this.strikeBonusA = 0;
+  this.strikeBonusB = 0;
+  this.spareBonusA = 0;
+  this.spareBonusB = 0;
+  this.knockedDown = [];
+};
+
+Game.prototype.play = function() {
+  if(this.frame < 10) {
+    this.knockedDown = [];
+    this.frameBall = 0;
+    this.rollBall();
+    this.bonusPoints();
+    this.strikeBonusA-=1;
+    this.strikeBonusB-=1;
+    this.spareBonusA-=1;
+    this.spareBonusB-=1;
+    this.scoreCard.push(this.knockedDown);
+    this.frame += 1;
+    if(this.frame==10){
+    this.eleventhFrame();
+    };
+  };
+  return this.frame;
+};
+
+Game.prototype.rollBall = function() {
+  while(this.frameBall < 2) {
+    this.frameBall+=1;
+    this.frameScore = this.rollScore();
+    this.updateScoreCard();
+    this.totalScore();
+  };
+  return this.frameScore
+};
+
+Game.prototype.rollScore = function() {
+  if(this.frameBall==1) {
+    this.firstRoll = Math.floor(Math.random()*11);
+    if(this.firstRoll==10) {
+      this.frameBall+=1;
+      if(this.strikeBonusA==1) {
+        this.strikeBonusB=2;
+      } else {
+        this.strikeBonusA=2;
+      }
+    }
+    return this.firstRoll
+   } else {
+     this.secondRoll = Math.floor(Math.random()*(11-this.firstRoll));
+     this.rollTotal = (this.frameScore+this.secondRoll);
+     if(this.rollTotal==10) {
+       if(this.spareBonusA==1) {
+        this.spareBonusB=2;
+      } else {
+        this.spareBonusA=2;
+      }
+     }
+     return this.secondRoll
+   };
+};
+
```

Comment:
Beware the pyramid of doom. Multiple if else statements are a sign that you should extract your logic into an object.